<launch>

    <arg name="vehicle" default="kiwi"/>
    <arg name="run_estimator" default="true"/>
    <arg name="run_verify" default="false"/>
    <arg name="run_odom_to_velocity" default="false"/>
    <arg name="run_uncc" default="false"/>
    <arg name="run_demo_rgbd" default="false"/>
    <arg name="run_mocap_vel" default="true"/>
    <arg name="play_bag" default="true"/>
    <arg name="run_plot_juggler" default="false"/>
    <arg name="run_image_view" default="false"/>
    <arg name="run_histogram" default="false"/>
    <arg name="record_bag" default="false"/>
    <arg name="bag_name" default="test"/>

    <rosparam command="load" file="$(find reef_estimator)/params/$(arg vehicle)_camera.yaml" />

    <group if="$(arg play_bag)">
<!--        <arg name="bag_path" default="/home/humberto/Downloads/adapative_control_bag_filtered.bag"/>-->
        <arg name="bag_path" default="/home/humberto/data_ardupilot_reef_estimator/pixhawk-90hz.bag"/>
        <node pkg="rosbag" type="play" name="player" output="screen" args="--clock --pause $(arg bag_path)"/>
    </group>

    <group if="$(arg run_plot_juggler)">
        <arg name="load_measurement_layout" default="true"/>
        <arg unless="$(arg load_measurement_layout)" name="layout_path" default="$(find reef_estimator)/config/sync_estimates_layout.xml"/>
        <arg if="$(arg load_measurement_layout)" name="layout_path" default="$(find reef_estimator)/config/sync_estimates_measurements_layout.xml"/>
        <node name="plotjuggler_with_layout" pkg="plotjuggler" type="PlotJuggler" args="--layout $(arg layout_path)"/>
    </group>

    <node if="$(arg run_estimator)" name="reef_estimator" pkg="reef_estimator" type="reef_estimator" output="screen">
        <rosparam file="$(find reef_estimator)/params/xy_est_params.yaml" />
        <rosparam file="$(find reef_estimator)/params/z_est_params.yaml" />
        <rosparam file="$(find reef_estimator)/params/basic_params.yaml" />
        <rosparam>
            enable_rgbd: false
            enable_sonar: false
            enable_mocap_xy: true
            enable_mocap_z: true
        </rosparam>
        <remap from="mocap_ned" to="/bugstomper/ned/pose_stamped"/>
        <remap from="mocap_velocity/body_level_frame" to="velocity/body_level_frame"/>
        <remap from="rgbd_velocity_body_frame" to="rgbd_to_velocity/body_level_frame"/>
        <remap from="imu/data" to="/mavros/imu/data"/>
    </node>


    <group if="$(arg run_demo_rgbd)">
        <include file="$(find demo_rgbd)/launch/demo_rgbd.launch"/>
        <include file="$(find rgbd_to_velocity)/launch/rgbd_to_velocity.launch"/>
    </group>


    <group if="$(arg run_uncc)">

        <node name="odom_to_velocity" pkg="odom_to_velocity" type="odom_to_velocity_node" output="screen">
            <rosparam command="load" file="$(find odom_to_velocity)/params/basic.yaml" />
        </node>

        <include if="$(arg run_uncc)" file="$(find uncc_rgbd_odom)/launch/astra_rgbd.launch">
            <arg name="rgb_topic" default="image_raw"/>
            <arg name="depth_topic" default="image_raw"/>
        </include>
    </group>


    <node if="$(arg run_mocap_vel)" name="pose_to_vel" pkg="position_to_velocity" type="position_to_velocity_node" output="screen" >
        <rosparam command="load" file="$(find position_to_velocity)/params/basic.yaml" />
        <remap from="pose_stamped" to="/bugstomper/ned/pose_stamped"/>
    </node>

    <node if="$(arg run_verify)" name="verify_estimates" pkg="reef_estimator" type="verify_estimates.py" output="screen">
        <rosparam>
            use_sonar: false
            SD_Multiplied: 1
            Z_offset: 0.0
        </rosparam>
        <remap from="mocap/velocity/body_level_frame" to="velocity/body_level_frame"/>
        <remap from="xyz_debug_estimates" to="xyz_debug_estimate"/>
    </node>

    <node if="$(arg run_histogram)" name="online_histogram" pkg="reef_estimator" type="online_histogram.py" output="screen"/>


    <node if="$(arg run_image_view)" name="image_view" pkg="image_view" type="image_view" output="screen">
        <remap from="image" to="camera/rgb/image_raw" />
    </node>

    <node if="$(arg record_bag)" name="record" pkg="rosbag" type="record" args="-O $(arg bag_name) sonar pose_stamped xyz_debug_estimate rgbd_velocity/body_level_frame velocity/body_level_frame"/>


</launch>
